Makefile

```diff
@@ -28,7 +28,9 @@ preflight-check:
 	$(srcdir)/preflight-check.sh
 
 python_code: setup.py
-	pip${python_version} install .
+	$(eval python_major_version := $(shell echo ${python_version} | cut -d '.' -f 1))
+	$(eval PIP ?= $(shell [ -x "$$(command -v pip${python_version})" ] && echo pip${python_version} || [ -x "$$(command -v pip${python_major_version})" ] && echo pip${python_major_version} || echo pip))
```

Comment:
I suggest that the `shell` part could be reduced to the equivalent of:

```
basename $(command -v pip3.11 || command -v pip3 || command -v pip)
```

Reply:
I agree that this looks cleaner. However, it may produce unexpected results if pip is an alias, take a look at this example:
```
$ command -v ls
alias ls='ls --color=auto'
$ basename $(command -v ls)
basename: extra operand ‘--color=auto'’
Try 'basename --help' for more information.
$
```